subsys/bluetooth/controller/ll_sw/ull_conn.c

```diff
@@ -5513,12 +5551,14 @@ static inline void ctrl_tx_ack(struct ll_conn *conn, struct node_tx **tx,
 	switch (pdu_tx->llctrl.opcode) {
 	case PDU_DATA_LLCTRL_TYPE_TERMINATE_IND:
 	{
-		uint8_t reason = (pdu_tx->llctrl.terminate_ind.error_code ==
-			       BT_HCI_ERR_REMOTE_USER_TERM_CONN) ?
-			      BT_HCI_ERR_LOCALHOST_TERM_CONN :
+		if (pdu_tx->llctrl.terminate_ind.error_code ==
+		    BT_HCI_ERR_REMOTE_USER_TERM_CONN) {
+			conn->llcp_terminate.reason_final =
+				BT_HCI_ERR_LOCALHOST_TERM_CONN;
+		} else {
+			conn->llcp_terminate.reason_final =
 			      pdu_tx->llctrl.terminate_ind.error_code;
-
-		conn_cleanup(conn, reason);
```

Comment:
It is not entirely clear to me where the corresponding call to `conn_cleanup` happens after the change. Is it from `ull_conn_done` (line 1228)?

Reply:
yes. https://github.com/zephyrproject-rtos/zephyr/pull/33367/files?file-filters%5B%5D=.c#diff-135b49c1c599aa1060145905101574cb58224b39926a1e1a04dda6ff6e8702a0R1232